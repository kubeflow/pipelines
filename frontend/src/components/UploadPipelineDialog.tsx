/*
 * Copyright 2018 The Kubeflow Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from 'react';
import BusyButton from '../atoms/BusyButton';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogTitle from '@material-ui/core/DialogTitle';
import Dropzone from 'react-dropzone';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Input from '../atoms/Input';
import InputAdornment from '@material-ui/core/InputAdornment';
import Radio from '@material-ui/core/Radio';
import { TextFieldProps } from '@material-ui/core/TextField';
import { padding, commonCss, zIndex, color } from '../Css';
import { stylesheet, classes } from 'typestyle';
import { ExternalLink } from '../atoms/ExternalLink';

const css = stylesheet({
  dropOverlay: {
    backgroundColor: color.lightGrey,
    border: '2px dashed #aaa',
    bottom: 0,
    left: 0,
    padding: '2.5em 0',
    position: 'absolute',
    right: 0,
    textAlign: 'center',
    top: 0,
    zIndex: zIndex.DROP_ZONE_OVERLAY,
  },
  root: {
    width: 500,
  },
});

export enum ImportMethod {
  LOCAL = 'local',
  URL = 'url',
}

interface UploadPipelineDialogProps {
  open: boolean;
  onClose: (
    confirmed: boolean,
    name: string,
    file: File | null,
    url: string,
    method: ImportMethod,
    description?: string,
  ) => Promise<boolean>;
}

interface UploadPipelineDialogState {
  busy: boolean;
  dropzoneActive: boolean;
  file: File | null;
  fileName: string;
  fileUrl: string;
  importMethod: ImportMethod;
  uploadPipelineDescription: string;
  uploadPipelineName: string;
}

class UploadPipelineDialog extends React.Component<
  UploadPipelineDialogProps,
  UploadPipelineDialogState
> {
  private _dropzoneRef = React.createRef<Dropzone & HTMLDivElement>();

  constructor(props: any) {
    super(props);

    this.state = {
      busy: false,
      dropzoneActive: false,
      file: null,
      fileName: '',
      fileUrl: '',
      importMethod: ImportMethod.LOCAL,
      uploadPipelineDescription: '',
      uploadPipelineName: '',
    };
  }

  public render(): JSX.Element {
    const {
      dropzoneActive,
      file,
      fileName,
      fileUrl,
      importMethod,
      uploadPipelineName,
      busy,
    } = this.state;

    return (
      <Dialog
        id='uploadDialog'
        onClose={() => this._uploadDialogClosed(false)}
        open={this.props.open}
        classes={{ paper: css.root }}
      >
        <DialogTitle>Upload and name your pipeline</DialogTitle>

        <div className={padding(20, 'lr')}>
          <div className={classes(commonCss.flex, padding(10, 'b'))}>
            <FormControlLabel
              id='uploadLocalFileBtn'
              label='Upload a file'
              checked={importMethod === ImportMethod.LOCAL}
              control={<Radio color='primary' />}
              onChange={() => this.setState({ importMethod: ImportMethod.LOCAL })}
            />
            <FormControlLabel
              id='uploadFromUrlBtn'
              label='Import by URL'
              checked={importMethod === ImportMethod.URL}
              control={<Radio color='primary' />}
              onChange={() => this.setState({ importMethod: ImportMethod.URL })}
            />
          </div>

          {importMethod === ImportMethod.LOCAL && (
            <React.Fragment>
              <Dropzone
                id='dropZone'
                disableClick={true}
                onDrop={this._onDrop.bind(this)}
                onDragEnter={this._onDropzoneDragEnter.bind(this)}
                onDragLeave={this._onDropzoneDragLeave.bind(this)}
                style={{ position: 'relative' }}
                ref={this._dropzoneRef}
                inputProps={{ tabIndex: -1 }}
              >
                {dropzoneActive && <div className={css.dropOverlay}>Drop files..</div>}

                <div className={padding(10, 'b')}>
                  Choose a pipeline package file from your computer, and give the pipeline a unique
                  name.
                  <br />
                  You can also drag and drop the file here.
                </div>
                <DocumentationCompilePipeline />
                <Input
                  onChange={this.handleChange('fileName')}
                  value={fileName}
                  required={true}
                  label='File'
                  variant='outlined'
                  InputProps={{
                    endAdornment: (
                      <InputAdornment position='end'>
                        <Button
                          color='secondary'
                          onClick={() => this._dropzoneRef.current!.open()}
                          style={{ padding: '3px 5px', margin: 0, whiteSpace: 'nowrap' }}
                        >
                          Choose file
                        </Button>
                      </InputAdornment>
                    ),
                    readOnly: true,
                  }}
                />
              </Dropzone>
            </React.Fragment>
          )}

          {importMethod === ImportMethod.URL && (
            <React.Fragment>
              <div className={padding(10, 'b')}>URL must be publicly accessible.</div>
              <DocumentationCompilePipeline />
              <Input
                onChange={this.handleChange('fileUrl')}
                value={fileUrl}
                required={true}
                label='URL'
                variant='outlined'
              />
            </React.Fragment>
          )}

          <Input
            id='uploadFileName'
            label='Pipeline name'
            onChange={this.handleChange('uploadPipelineName')}
            required={true}
            value={uploadPipelineName}
            variant='outlined'
          />
        </div>

        {/* <Input label='Pipeline description'
          onChange={this.handleChange('uploadPipelineDescription')}
          value={uploadPipelineDescription} multiline={true} variant='outlined' /> */}

        <DialogActions>
          <Button id='cancelUploadBtn' onClick={() => this._uploadDialogClosed.bind(this)(false)}>
            Cancel
          </Button>
          <BusyButton
            id='confirmUploadBtn'
            onClick={() => this._uploadDialogClosed.bind(this)(true)}
            title='Upload'
            busy={busy}
            disabled={
              !uploadPipelineName || (importMethod === ImportMethod.LOCAL ? !file : !fileUrl)
            }
          />
        </DialogActions>
      </Dialog>
    );
  }

  public handleChange = (name: string) => (event: any) => {
    this.setState({
      [name]: (event.target as TextFieldProps).value,
    } as any);
  };

  private _onDropzoneDragEnter(): void {
    this.setState({ dropzoneActive: true });
  }

  private _onDropzoneDragLeave(): void {
    this.setState({ dropzoneActive: false });
  }

  private _onDrop(files: File[]): void {
    this.setState({
      dropzoneActive: false,
      file: files[0],
      fileName: files[0].name,
      // Suggest all characters left of first . as pipeline name
      uploadPipelineName: files[0].name.split('.')[0],
    });
  }

  private _uploadDialogClosed(confirmed: boolean): void {
    this.setState({ busy: true }, async () => {
      const success = await this.props.onClose(
        confirmed,
        this.state.uploadPipelineName,
        this.state.file,
        this.state.fileUrl.trim(),
        this.state.importMethod,
        this.state.uploadPipelineDescription,
      );
      if (success) {
        this.setState({
          busy: false,
          dropzoneActive: false,
          file: null,
          fileName: '',
          fileUrl: '',
          importMethod: ImportMethod.LOCAL,
          uploadPipelineDescription: '',
          uploadPipelineName: '',
        });
      } else {
        this.setState({ busy: false });
      }
    });
  }
}

export default UploadPipelineDialog;

export const DocumentationCompilePipeline: React.FC = () => (
  <div className={padding(10, 'b')}>
    For expected file format, refer to{' '}
    <ExternalLink href='https://www.kubeflow.org/docs/components/pipelines/sdk/v2/build-pipeline/#compile-and-run-your-pipeline'>
      Compile Pipeline Documentation
    </ExternalLink>
    .
  </div>
);
