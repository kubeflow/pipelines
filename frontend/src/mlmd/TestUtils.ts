/*
 * Copyright 2018,2021 The Kubeflow Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Artifact, Value } from 'src/third_party/mlmd';
import { ArtifactCustomProperties, ArtifactProperties } from './Api';

export const doubleValue = (number: number) => {
  const value = new Value();
  value.setDoubleValue(number);
  return value;
};

export const intValue = (number: number) => {
  const value = new Value();
  value.setIntValue(number);
  return value;
};

export const stringValue = (string: String) => {
  const value = new Value();
  value.setStringValue(String(string));
  return value;
};

export const buildTestModel = () => {
  const model = new Artifact();
  model.setId(1);
  model.setTypeId(1);
  model.setUri('gs://my-bucket/mnist');
  model.getPropertiesMap().set(ArtifactProperties.NAME, stringValue('test model'));
  model.getPropertiesMap().set(ArtifactProperties.DESCRIPTION, stringValue('A really great model'));
  model.getPropertiesMap().set(ArtifactProperties.VERSION, stringValue('v1'));
  model
    .getPropertiesMap()
    .set(ArtifactProperties.CREATE_TIME, stringValue('2019-06-12T01:21:48.259263Z'));
  model
    .getPropertiesMap()
    .set(
      ArtifactProperties.ALL_META,
      stringValue(
        '{"hyperparameters": {"early_stop": true, ' +
          '"layers": [10, 3, 1], "learning_rate": 0.5}, ' +
          '"model_type": "neural network", ' +
          '"training_framework": {"name": "tensorflow", "version": "v1.0"}}',
      ),
    );
  model
    .getCustomPropertiesMap()
    .set(ArtifactCustomProperties.WORKSPACE, stringValue('workspace-1'));
  model.getCustomPropertiesMap().set(ArtifactCustomProperties.RUN, stringValue('1'));
  return model;
};

export const testModel = buildTestModel();
