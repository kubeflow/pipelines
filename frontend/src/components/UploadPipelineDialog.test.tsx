/*
 * Copyright 2018 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from 'react';
import { shallow, ReactWrapper, ShallowWrapper } from 'enzyme';
import UploadPipelineDialog, { ImportMethod } from './UploadPipelineDialog';
import TestUtils from '../TestUtils';

describe('UploadPipelineDialog', () => {
  let tree: ReactWrapper | ShallowWrapper;

  afterEach(async () => {
    // unmount() should be called before resetAllMocks() in case any part of the unmount life cycle
    // depends on mocks/spies
    await tree.unmount();
  });

  it('renders closed', () => {
    tree = shallow(<UploadPipelineDialog open={false} onClose={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders open', () => {
    tree = shallow(<UploadPipelineDialog open={false} onClose={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders an active dropzone', () => {
    tree = shallow(<UploadPipelineDialog open={false} onClose={jest.fn()} />);
    tree.setState({ dropzoneActive: true });
    expect(tree).toMatchSnapshot();
  });

  it('renders with a selected file to upload', () => {
    tree = shallow(<UploadPipelineDialog open={false} onClose={jest.fn()} />);
    tree.setState({ fileToUpload: true });
    expect(tree).toMatchSnapshot();
  });

  it('renders alternate UI for uploading via URL', () => {
    tree = shallow(<UploadPipelineDialog open={false} onClose={jest.fn()} />);
    tree.setState({ importMethod: ImportMethod.URL });
    expect(tree).toMatchSnapshot();
  });

  it('calls close callback with null and empty string when canceled', () => {
    const spy = jest.fn();
    tree = shallow(<UploadPipelineDialog open={false} onClose={spy} />);
    tree.find('#cancelUploadBtn').simulate('click');
    expect(spy).toHaveBeenCalledWith(false, '', null, '', ImportMethod.LOCAL, '');
  });

  it('calls close callback with null and empty string when dialog is closed', () => {
    const spy = jest.fn();
    tree = shallow(<UploadPipelineDialog open={false} onClose={spy} />);
    tree.find('WithStyles(Dialog)').simulate('close');
    expect(spy).toHaveBeenCalledWith(false, '', null, '', ImportMethod.LOCAL, '');
  });

  it('calls close callback with file name, file object, and description when confirmed', () => {
    const spy = jest.fn();
    tree = shallow(<UploadPipelineDialog open={false} onClose={spy} />);
    (tree.instance() as any)._dropzoneRef = { current: { open: () => null } };
    (tree.instance() as UploadPipelineDialog).handleChange('uploadPipelineName')({
      target: { value: 'test name' },
    });
    tree.find('#confirmUploadBtn').simulate('click');
    expect(spy).toHaveBeenLastCalledWith(true, 'test name', null, '', ImportMethod.LOCAL, '');
  });

  it('calls close callback with trimmed file url and pipeline name when confirmed', () => {
    const spy = jest.fn();
    tree = shallow(<UploadPipelineDialog open={false} onClose={spy} />);
    // Click 'Import by URL'
    tree.find('#uploadFromUrlBtn').simulate('change');
    (tree.instance() as UploadPipelineDialog).handleChange('fileUrl')({
      target: { value: '\n https://www.google.com/test-file.txt ' },
    });
    (tree.instance() as UploadPipelineDialog).handleChange('uploadPipelineName')({
      target: { value: 'test name' },
    });
    tree.find('#confirmUploadBtn').simulate('click');
    expect(spy).toHaveBeenLastCalledWith(
      true,
      'test name',
      null,
      'https://www.google.com/test-file.txt',
      ImportMethod.URL,
      '',
    );
  });

  it('trims file extension for pipeline name suggestion', () => {
    tree = shallow(<UploadPipelineDialog open={false} onClose={jest.fn()} />);
    const file = { name: 'test_upload_file.tar.gz' };
    tree.find('#dropZone').simulate('drop', [file]);
    expect(tree.state()).toHaveProperty('dropzoneActive', false);
    expect(tree.state()).toHaveProperty('uploadPipelineName', 'test_upload_file');
  });

  it('sets the import method based on which radio button is toggled', () => {
    tree = shallow(<UploadPipelineDialog open={false} onClose={jest.fn()} />);
    // Import method is LOCAL by default
    expect(tree.state('importMethod')).toBe(ImportMethod.LOCAL);

    // Click 'Import by URL'
    tree.find('#uploadFromUrlBtn').simulate('change');
    expect(tree.state('importMethod')).toBe(ImportMethod.URL);

    // Click back to default, 'Upload a file'
    tree.find('#uploadLocalFileBtn').simulate('change');
    expect(tree.state('importMethod')).toBe(ImportMethod.LOCAL);
  });

  it('resets all state if the dialog is closed and the callback returns true', async () => {
    const spy = jest.fn(() => true);

    tree = shallow(<UploadPipelineDialog open={false} onClose={spy} />);
    tree.setState({
      busy: true,
      dropzoneActive: true,
      file: {},
      fileName: 'test file name',
      fileUrl: 'https://some.url.com',
      importMethod: ImportMethod.URL,
      uploadPipelineDescription: 'test description',
      uploadPipelineName: 'test pipeline name',
    });

    tree.find('#confirmUploadBtn').simulate('click');
    await TestUtils.flushPromises();

    expect(tree.state('busy')).toBe(false);
    expect(tree.state('dropzoneActive')).toBe(false);
    expect(tree.state('file')).toBeNull();
    expect(tree.state('fileName')).toBe('');
    expect(tree.state('fileUrl')).toBe('');
    expect(tree.state('importMethod')).toBe(ImportMethod.LOCAL);
    expect(tree.state('uploadPipelineDescription')).toBe('');
    expect(tree.state('uploadPipelineName')).toBe('');
  });

  it('does not reset the state if the dialog is closed and the callback returns false', async () => {
    const spy = jest.fn(() => false);

    tree = shallow(<UploadPipelineDialog open={false} onClose={spy} />);
    tree.setState({
      busy: true,
      dropzoneActive: true,
      file: {},
      fileName: 'test file name',
      fileUrl: 'https://some.url.com',
      importMethod: ImportMethod.URL,
      uploadPipelineDescription: 'test description',
      uploadPipelineName: 'test pipeline name',
    });

    tree.find('#confirmUploadBtn').simulate('click');
    await TestUtils.flushPromises();

    expect(tree.state('dropzoneActive')).toBe(true);
    expect(tree.state('file')).toEqual({});
    expect(tree.state('fileName')).toBe('test file name');
    expect(tree.state('fileUrl')).toBe('https://some.url.com');
    expect(tree.state('importMethod')).toBe(ImportMethod.URL);
    expect(tree.state('uploadPipelineDescription')).toBe('test description');
    expect(tree.state('uploadPipelineName')).toBe('test pipeline name');
    // 'busy' is set to false regardless upon the callback returning
    expect(tree.state('busy')).toBe(false);
  });

  it('sets an active dropzone on drag', () => {
    tree = shallow(<UploadPipelineDialog open={false} onClose={jest.fn()} />);
    tree.find('#dropZone').simulate('dragEnter');
    expect(tree.state()).toHaveProperty('dropzoneActive', true);
  });

  it('sets an inactive dropzone on drag leave', () => {
    tree = shallow(<UploadPipelineDialog open={false} onClose={jest.fn()} />);
    tree.find('#dropZone').simulate('dragLeave');
    expect(tree.state()).toHaveProperty('dropzoneActive', false);
  });

  it('sets a file object on drop', () => {
    tree = shallow(<UploadPipelineDialog open={false} onClose={jest.fn()} />);
    const file = { name: 'test upload file' };
    tree.find('#dropZone').simulate('drop', [file]);
    expect(tree.state()).toHaveProperty('dropzoneActive', false);
    expect(tree.state()).toHaveProperty('uploadPipelineName', file.name);
  });
});
