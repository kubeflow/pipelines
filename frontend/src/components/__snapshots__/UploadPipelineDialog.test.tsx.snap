// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadPipelineDialog renders alternate UI for uploading via URL 1`] = `
<WithStyles(Dialog)
  classes={
    Object {
      "paper": "root",
    }
  }
  id="uploadDialog"
  onClose={[Function]}
  open={false}
>
  <WithStyles(DialogTitle)>
    Upload and name your pipeline
  </WithStyles(DialogTitle)>
  <div
    className=""
  >
    <div
      className="flex"
    >
      <WithStyles(WithFormControlContext(FormControlLabel))
        checked={false}
        control={
          <WithStyles(Radio)
            color="primary"
          />
        }
        id="uploadLocalFileBtn"
        label="Upload a file"
        onChange={[Function]}
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        checked={true}
        control={
          <WithStyles(Radio)
            color="primary"
          />
        }
        id="uploadFromUrlBtn"
        label="Import by URL"
        onChange={[Function]}
      />
    </div>
    <div
      className=""
    >
      URL must be publicly accessible.
    </div>
    <DocumentationCompilePipeline />
    <Input
      label="URL"
      onChange={[Function]}
      required={true}
      value=""
      variant="outlined"
    />
    <Input
      id="uploadFileName"
      label="Pipeline name"
      onChange={[Function]}
      required={true}
      value=""
      variant="outlined"
    />
  </div>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      id="cancelUploadBtn"
      onClick={[Function]}
    >
      Cancel
    </WithStyles(Button)>
    <BusyButton
      busy={false}
      disabled={true}
      id="confirmUploadBtn"
      onClick={[Function]}
      title="Upload"
    />
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;

exports[`UploadPipelineDialog renders an active dropzone 1`] = `
<WithStyles(Dialog)
  classes={
    Object {
      "paper": "root",
    }
  }
  id="uploadDialog"
  onClose={[Function]}
  open={false}
>
  <WithStyles(DialogTitle)>
    Upload and name your pipeline
  </WithStyles(DialogTitle)>
  <div
    className=""
  >
    <div
      className="flex"
    >
      <WithStyles(WithFormControlContext(FormControlLabel))
        checked={true}
        control={
          <WithStyles(Radio)
            color="primary"
          />
        }
        id="uploadLocalFileBtn"
        label="Upload a file"
        onChange={[Function]}
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        checked={false}
        control={
          <WithStyles(Radio)
            color="primary"
          />
        }
        id="uploadFromUrlBtn"
        label="Import by URL"
        onChange={[Function]}
      />
    </div>
    <n
      disableClick={true}
      disablePreview={false}
      disabled={false}
      getDataTransferItems={[Function]}
      id="dropZone"
      inputProps={
        Object {
          "tabIndex": -1,
        }
      }
      maxSize={Infinity}
      minSize={0}
      multiple={true}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDrop={[Function]}
      preventDropOnDocument={true}
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        className="dropOverlay"
      >
        Drop files..
      </div>
      <div
        className=""
      >
        Choose a pipeline package file from your computer, and give the pipeline a unique name.
        <br />
        You can also drag and drop the file here.
      </div>
      <DocumentationCompilePipeline />
      <Input
        InputProps={
          Object {
            "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
              position="end"
            >
              <WithStyles(Button)
                color="secondary"
                onClick={[Function]}
                style={
                  Object {
                    "margin": 0,
                    "padding": "3px 5px",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                Choose file
              </WithStyles(Button)>
            </WithStyles(WithFormControlContext(InputAdornment))>,
            "readOnly": true,
          }
        }
        label="File"
        onChange={[Function]}
        required={true}
        value=""
        variant="outlined"
      />
    </n>
    <Input
      id="uploadFileName"
      label="Pipeline name"
      onChange={[Function]}
      required={true}
      value=""
      variant="outlined"
    />
  </div>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      id="cancelUploadBtn"
      onClick={[Function]}
    >
      Cancel
    </WithStyles(Button)>
    <BusyButton
      busy={false}
      disabled={true}
      id="confirmUploadBtn"
      onClick={[Function]}
      title="Upload"
    />
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;

exports[`UploadPipelineDialog renders closed 1`] = `
<WithStyles(Dialog)
  classes={
    Object {
      "paper": "root",
    }
  }
  id="uploadDialog"
  onClose={[Function]}
  open={false}
>
  <WithStyles(DialogTitle)>
    Upload and name your pipeline
  </WithStyles(DialogTitle)>
  <div
    className=""
  >
    <div
      className="flex"
    >
      <WithStyles(WithFormControlContext(FormControlLabel))
        checked={true}
        control={
          <WithStyles(Radio)
            color="primary"
          />
        }
        id="uploadLocalFileBtn"
        label="Upload a file"
        onChange={[Function]}
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        checked={false}
        control={
          <WithStyles(Radio)
            color="primary"
          />
        }
        id="uploadFromUrlBtn"
        label="Import by URL"
        onChange={[Function]}
      />
    </div>
    <n
      disableClick={true}
      disablePreview={false}
      disabled={false}
      getDataTransferItems={[Function]}
      id="dropZone"
      inputProps={
        Object {
          "tabIndex": -1,
        }
      }
      maxSize={Infinity}
      minSize={0}
      multiple={true}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDrop={[Function]}
      preventDropOnDocument={true}
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        className=""
      >
        Choose a pipeline package file from your computer, and give the pipeline a unique name.
        <br />
        You can also drag and drop the file here.
      </div>
      <DocumentationCompilePipeline />
      <Input
        InputProps={
          Object {
            "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
              position="end"
            >
              <WithStyles(Button)
                color="secondary"
                onClick={[Function]}
                style={
                  Object {
                    "margin": 0,
                    "padding": "3px 5px",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                Choose file
              </WithStyles(Button)>
            </WithStyles(WithFormControlContext(InputAdornment))>,
            "readOnly": true,
          }
        }
        label="File"
        onChange={[Function]}
        required={true}
        value=""
        variant="outlined"
      />
    </n>
    <Input
      id="uploadFileName"
      label="Pipeline name"
      onChange={[Function]}
      required={true}
      value=""
      variant="outlined"
    />
  </div>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      id="cancelUploadBtn"
      onClick={[Function]}
    >
      Cancel
    </WithStyles(Button)>
    <BusyButton
      busy={false}
      disabled={true}
      id="confirmUploadBtn"
      onClick={[Function]}
      title="Upload"
    />
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;

exports[`UploadPipelineDialog renders open 1`] = `
<WithStyles(Dialog)
  classes={
    Object {
      "paper": "root",
    }
  }
  id="uploadDialog"
  onClose={[Function]}
  open={false}
>
  <WithStyles(DialogTitle)>
    Upload and name your pipeline
  </WithStyles(DialogTitle)>
  <div
    className=""
  >
    <div
      className="flex"
    >
      <WithStyles(WithFormControlContext(FormControlLabel))
        checked={true}
        control={
          <WithStyles(Radio)
            color="primary"
          />
        }
        id="uploadLocalFileBtn"
        label="Upload a file"
        onChange={[Function]}
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        checked={false}
        control={
          <WithStyles(Radio)
            color="primary"
          />
        }
        id="uploadFromUrlBtn"
        label="Import by URL"
        onChange={[Function]}
      />
    </div>
    <n
      disableClick={true}
      disablePreview={false}
      disabled={false}
      getDataTransferItems={[Function]}
      id="dropZone"
      inputProps={
        Object {
          "tabIndex": -1,
        }
      }
      maxSize={Infinity}
      minSize={0}
      multiple={true}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDrop={[Function]}
      preventDropOnDocument={true}
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        className=""
      >
        Choose a pipeline package file from your computer, and give the pipeline a unique name.
        <br />
        You can also drag and drop the file here.
      </div>
      <DocumentationCompilePipeline />
      <Input
        InputProps={
          Object {
            "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
              position="end"
            >
              <WithStyles(Button)
                color="secondary"
                onClick={[Function]}
                style={
                  Object {
                    "margin": 0,
                    "padding": "3px 5px",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                Choose file
              </WithStyles(Button)>
            </WithStyles(WithFormControlContext(InputAdornment))>,
            "readOnly": true,
          }
        }
        label="File"
        onChange={[Function]}
        required={true}
        value=""
        variant="outlined"
      />
    </n>
    <Input
      id="uploadFileName"
      label="Pipeline name"
      onChange={[Function]}
      required={true}
      value=""
      variant="outlined"
    />
  </div>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      id="cancelUploadBtn"
      onClick={[Function]}
    >
      Cancel
    </WithStyles(Button)>
    <BusyButton
      busy={false}
      disabled={true}
      id="confirmUploadBtn"
      onClick={[Function]}
      title="Upload"
    />
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;

exports[`UploadPipelineDialog renders with a selected file to upload 1`] = `
<WithStyles(Dialog)
  classes={
    Object {
      "paper": "root",
    }
  }
  id="uploadDialog"
  onClose={[Function]}
  open={false}
>
  <WithStyles(DialogTitle)>
    Upload and name your pipeline
  </WithStyles(DialogTitle)>
  <div
    className=""
  >
    <div
      className="flex"
    >
      <WithStyles(WithFormControlContext(FormControlLabel))
        checked={true}
        control={
          <WithStyles(Radio)
            color="primary"
          />
        }
        id="uploadLocalFileBtn"
        label="Upload a file"
        onChange={[Function]}
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        checked={false}
        control={
          <WithStyles(Radio)
            color="primary"
          />
        }
        id="uploadFromUrlBtn"
        label="Import by URL"
        onChange={[Function]}
      />
    </div>
    <n
      disableClick={true}
      disablePreview={false}
      disabled={false}
      getDataTransferItems={[Function]}
      id="dropZone"
      inputProps={
        Object {
          "tabIndex": -1,
        }
      }
      maxSize={Infinity}
      minSize={0}
      multiple={true}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDrop={[Function]}
      preventDropOnDocument={true}
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        className=""
      >
        Choose a pipeline package file from your computer, and give the pipeline a unique name.
        <br />
        You can also drag and drop the file here.
      </div>
      <DocumentationCompilePipeline />
      <Input
        InputProps={
          Object {
            "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
              position="end"
            >
              <WithStyles(Button)
                color="secondary"
                onClick={[Function]}
                style={
                  Object {
                    "margin": 0,
                    "padding": "3px 5px",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                Choose file
              </WithStyles(Button)>
            </WithStyles(WithFormControlContext(InputAdornment))>,
            "readOnly": true,
          }
        }
        label="File"
        onChange={[Function]}
        required={true}
        value=""
        variant="outlined"
      />
    </n>
    <Input
      id="uploadFileName"
      label="Pipeline name"
      onChange={[Function]}
      required={true}
      value=""
      variant="outlined"
    />
  </div>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      id="cancelUploadBtn"
      onClick={[Function]}
    >
      Cancel
    </WithStyles(Button)>
    <BusyButton
      busy={false}
      disabled={true}
      id="confirmUploadBtn"
      onClick={[Function]}
      title="Upload"
    />
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
